mycore-base/src/main/java/org/mycore/datamodel/metadata/MCRObjectService.java

```diff
@@ -840,4 +927,371 @@ public final int getFlagIndex(String value) {
+        return getClassificationsAsMCRMetaClassification(classId).stream()
+            .map(c -> new MCRCategoryID(c.getClassId(), c.getCategId()))
+            .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    /**
+     * This method get a single classification from the classification list as a string.
+     *
+     * @exception IndexOutOfBoundsException
+     *                throw this exception, if the index is invalid
+     * @return a classification string
+     */
+    public final String getClassification(int index) throws IndexOutOfBoundsException {
+        if (index < 0 || index >= classifications.size()) {
+            throw new IndexOutOfBoundsException("Index error in getClassification.");
+        }
+        return classifications.get(index).getCategId();
+    }
+
+    /**
+     * This method gets a single message type from the message list as a string.
+     *
+     * @exception IndexOutOfBoundsException
+     *                throw this exception, if the index is invalid
+     * @return a message type
+     */
+    public final String getClassificationClassId(int index) throws IndexOutOfBoundsException {
+        if (index < 0 || index >= classifications.size()) {
+            throw new IndexOutOfBoundsException("Index error in getClassificationClassId.");
+        }
+        return classifications.get(index).getClassId();
+    }
+
+    /**
+     * This method get a single classification from the classification list as.
+     *
+     * @exception IndexOutOfBoundsException
+     *                throw this exception, if the index is invalid
+     * @return a classification value
+     */
+    public final MCRMetaClassification getClassificationAsMCRMetaClassification(int index)
+        throws IndexOutOfBoundsException {
+        if (index < 0 || index >= classifications.size()) {
+            throw new IndexOutOfBoundsException("Index error in getClassificationAsMCRMetaClassification.");
+        }
+        return classifications.get(index);
+    }
+
+
+    /**
+     * This method adds a classification to the classification list.
+     *
+     * @param classId
+     *              a cId as string
+     * @param categId
+     *              the new class id as string
+     */
+    public final void addClassification(String classId, String categId) {
+        MCRUtils.filterTrimmedNotEmpty(classId).ifPresent(classIdValue ->
+            MCRUtils.filterTrimmedNotEmpty(categId)
+                .map(categIdValue ->
+                    new MCRMetaClassification("servclass", 0, null, new MCRCategoryID(classIdValue, categIdValue)))
+                .ifPresent(classifications::add));
+    }
+
+    /**
+     * This method adds a classification to the classifications list.
+     *
+     * @param classification
+     *              the new classification value as string
+     */
+    public final void addClassification(MCRCategoryID classification) {
+        if (classification == null) {
```

Comment:
do the actual "adding" in this method and call this method in addClassfication(String, String)